### PR TITLE
docs: add contribution guide

### PR DESCRIPTION
**Pull request by mtrimolet**
Create documents to guide contributions to the project.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+```sh
+$ npm run lint 
+```
+
+It's also a good idea to hook up your editor to an eslint plugin. 
+
+To fix lint errors from the command line: 
+
+```sh
+$ npm run lint -- --fix
```
Comment (SomaticIT) [inline CONTRIBUTING.md:88]: - Does not work.
- Add a `lint:fix` shortcut task

**Comment by dradetsky** [comment]
this seems to work for me fwiw

**Comment by dradetsky** [comment]
Trying to follow the instructions for local CLI testing not going so well. Starting from a fresh clone of the repo (cloned from the local fs, but hopefully that's not relevant), I tried:

```
npm  ci
npx lerna bootstrap
npx lerna build
cd packages/cli
npm link
```

This fails with

```
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN tar ENOENT: no such file or directory, open '/home/dmr/git/lang/js/extra/clean/packages/cli/node_modules/.staging/yargs-6e1c9ba4/build/lib/apply-extends.d.ts'
npm WARN tar ENOENT: no such file or directory, open '/home/dmr/git/lang/js/extra/clean/packages/cli/node_modules/.staging/yargs-6e1c9ba4/build/lib/argsert.d.ts'
<snip like 40 lines>
npm WARN tar ENOENT: no such file or directory, open '/home/dmr/git/lang/js/extra/clean/packages/cli/node_modules/.staging/@jsdevtools/ono-a83dfb58/esm/types.d.ts'
npm ERR! cb() never called!

npm ERR! This is an error with npm itself. Please report this error at:
npm ERR!     <https://npm.community>
```

Wondering if this was maybe related to having done the build step, I tried a second fresh clone and just did

```
cd packages/cli
npm link
```

which failed with

```
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^2.1.2 (node_modules/jest-haste-map/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

added 640 packages from 380 contributors and audited 643 packages in 12.523s

22 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

npm ERR! code ENOENT
npm ERR! syscall chmod
npm ERR! path /home/dmr/.npm/lib/node_modules/@spec2ts/cli/bin/spec2ts.js
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, chmod '/home/dmr/.npm/lib/node_modules/@spec2ts/cli/bin/spec2ts.js'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/dmr/.npm/_logs/2020-10-07T22_37_24_919Z-debug.log
```

I can paste those logfiles somewhere if you think they are relevant.

Fun fact: I'm using npm 6.14.7.

**Comment by dradetsky** [comment]
Just for fun I also tried (in a 3rd fresh clone)

```
npm ci
cd packages/cli
npm link
```

which failed the same way as the 2nd attempt

**Comment by dradetsky** [comment]
Just to be clear, I don't mean to sound ungrateful. Thanks again to @mtrimolet for doing this. Even if it doesn't work for me 100% it's already a huge step in the right direction. I now know what I'm supposed to debug.

**Comment by mtrimolet** [comment]
I'm having the same issues with the cli testing procedure but I'm trying to figure out how to make it work from `npm link`.
Maybe we should stick to the "simple" procedure I'm using atm: 
```sh
lerna run build
cd packages/cli
node ./bin/spec2ts.js openapi ../openapi/tests/assets/petstore.yml -o .

# without build using ts-node, prefix with npx or install globally
ts-node ./bin/spec2ts.ts openapi ../openapi/tests/assets/petstore.yml -o .
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (mtrimolet) [reply CONTRIBUTING.md]: Is it working since e339a47 ?

**Review by SomaticIT — COMMENTED** [review]
Missing ISSUE_TEMPLATE.md

**Pull request merged**